apps/scripts/AliveTest.c

```diff
@@ -0,0 +1,143 @@
+// 4KB of D_MEM
+// 0x400800 - 0x400fff - Shared
+//
+// 0x400600 - 0x400800 - Thread 3
+// 0x400400 - 0x400600 - Thread 2
+// 0x400200 - 0x400400 - Thread 1
+// 0x400000 - 0x400200 - Thread 0
+
+// REGION == 2'b01;
+
+#define D_MEM_SHARED ((volatile int *) (0x00400f00))
+
+//////////Register-Immediate Instructions///////////
+void ADDI(int a){
+    D_MEM_SHARED[0] = a+1;
+}
+void SLTI(int a){
+    D_MEM_SHARED[1] = a<5;
+}
+void ANDI(int a){
+    D_MEM_SHARED[2] = a&6;
+}
+void ORI(int a){
+    a=a|26;
+    if(a==30)
+    	D_MEM_SHARED[3] = 1;
+}
+void XORI(int a){
+    D_MEM_SHARED[4] = a^42;
+}
+void SLLI(int a){
+    a=a<<3;
+    if(a==64)
+    	D_MEM_SHARED[5] = 1;
+}
+void SRLI(int a){
+    __asm__("lui a5,0x41;"
+    "addi a5,a5,-512;"
+    "lw a4,-20(s0);"
+    "srli a4,a4,0xc;"
+    "sw	a4,24(a5)");
+}
+
+void SRAI(int a){
+    a=a>>3;
+    if(a==-1024)
+    	D_MEM_SHARED[7] = 1;
+}
+
+//////////////////Register-Register Instructions//////////////////
+void ADD(int a, int b){
+    a=a+b;
+    if(a==579)
+    	D_MEM_SHARED[8] = 1;
+}
+
+
+void SLT(int a, int b){
+	D_MEM_SHARED[9]= a<b ? 1 : 0;
+} 
+
+/////////////////////////////////////
+void SLTU(int a){
+	D_MEM_SHARED[10]= a==0 ? 0 : 1;
```

Comment:
this expression may translate into stlu, but it's not that straightforward. perhaps you can use a signature like slt, but with unsigned ints:
```c
void SLTU(unsigned int a, unsigned int b)
```